packages/components/src/tabs/test/index.tsx

```diff
@@ -1278,4 +1285,60 @@ describe( 'Tabs', () => {
 			} );
 		} );
 	} );
+	it( 'should associate each `Tab` with the correct `TabPanel`, even if they are not rendered in the same order', async () => {
+		const TABS_WITH_DELTA_REVERSED = [ ...TABS_WITH_DELTA ].reverse();
+
+		render(
+			<Tabs>
+				<Tabs.TabList>
+					{ TABS_WITH_DELTA.map( ( tabObj ) => (
+						<Tabs.Tab
+							key={ tabObj.tabId }
+							tabId={ tabObj.tabId }
+							className={ tabObj.tab.className }
+							disabled={ tabObj.tab.disabled }
+						>
+							{ tabObj.title }
+						</Tabs.Tab>
+					) ) }
+				</Tabs.TabList>
+				{ TABS_WITH_DELTA_REVERSED.map( ( tabObj ) => (
+					<Tabs.TabPanel
+						key={ tabObj.tabId }
+						tabId={ tabObj.tabId }
+						focusable={ tabObj.tabpanel?.focusable }
+					>
+						{ tabObj.content }
+					</Tabs.TabPanel>
+				) ) }
+			</Tabs>
+		);
+
+		// Alpha is the initially selected tab,and should render the correct tabpanel
+		expect( await getSelectedTab() ).toHaveTextContent( 'Alpha' );
```

Comment:
I realize this change would go a bit beyond this test, but I wonder if it would be beneficial to add a parameter to the helper function so the assertions could be more specific/flexible. Like:

```diff
-const getSelectedTab = async () =>
-	await screen.findByRole( 'tab', { selected: true } );
+const getSelectedTab = async ( name: string ) =>
+	await screen.findByRole( 'tab', { name, selected: true } );
 ```

So then assertions could look like: 
 
```diff
-		expect( await getSelectedTab() ).toHaveTextContent( 'Alpha' );
+		expect( await getSelectedTab( 'Alpha' ) ).toBeVisible();
```

Reply:
Maybe worth a followup PR. If we went this route, I think it'd also make sense to rename the helper function to something like `confirmSelectedTab()`.

I also wouldn't want to transition to a `toBeVisible()` check. Because these are the actual tabs (not the tabpanels) they should all be visible whether they're currently selected or not. Definitely worth thinking about some more, thanks!

Reply:
>I also wouldn't want to transition to a toBeVisible() check. Because these are the actual tabs (not the tabpanels) they should all be visible whether they're currently selected or not. 

It's funny because I was thinking the opposite! 😄 By asserting `toBeVisible` for something that is always meant to be visible, we are ensuring that this important aspect is covered in the test. And it still checks for text content, by including the text content (`name`) in the query. 

From my understanding, I don't think `toBeVisible()` is meant to only check for things that are conditionally visible. I believe it is a more robust check to ensure it's appearing as expected for the user. [It covers things](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tobevisible) like `hidden` and `display: none;` and other properties that could be added to a Tab.